Implemented on-demand session routes

U-DON opened the pull request:
Resolves department-of-veterans-affairs/vets.gov-team#8888.
Depends on department-of-veterans-affairs/vets-api#1781.

Removes all of the old code that fetched URLs upfront and instead only fetch when the relevant links are clicked. The responses would then be loaded in the pop-up window.

U-DON committed:
Created helpers.

U-DON committed:
Used redirect for logout action.

U-DON committed:
Updated login routes.

U-DON committed:
Set SAML URL in popup after being fetched.

U-DON committed:
Modified signup helper.

U-DON committed:
Incorporated verify helper and removed obsolete code in Verify.

U-DON committed:
Refactored Signin.

U-DON committed:
Removed unused code.

U-DON committed:
Removed unused handleVerify.

U-DON committed:
Replaced old multifactor helpers.

U-DON committed:
Merge branch 'master' into login_routes

U-DON committed:
Removed unused getVerifyUrl.

U-DON committed:
Removed getLoginUrls helper.

U-DON committed:
Rearranged imports.

U-DON committed:
Removed obsolete URL actions.

U-DON committed:
Removed unused action.

U-DON committed:
Removed addEvent helper and used addEventListener where possible.

U-DON committed:
Removed prop and directly used helper function.

U-DON committed:
Removed tests for removed components.

U-DON committed:
Corrected Analytics events.

jbalboni commented:
@U-DON This looks awesome! I tried to checkout both this and the backend branch, but the front end just hangs on a loading spinner when I click on Sign In. Anything else I need to do to get this to run locally?

U-DON commented:
I'm just using this vets-website branch (`login_routes`) and the `redirect_url_json` branch for vets-api.

U-DON commented:
Oh, I see, there are still lingering mentions of `loginUrls`.

U-DON committed:
Removed remaining references to loginUrls.

U-DON commented:
Resolved. There should no longer be a spinner.

U-DON committed:
More error handling cases.

U-DON committed:
Return promises from popup helpers.

U-DON committed:
Replaced old login helpers test.

U-DON commented:
Added tests for the new helpers.

Added some error handling:
- If URL failed to be fetched, we open the generic login error in the popup window. Kind of hijacked the failure case for the SAML auth flow for this...
- If failed to open a window at all, we report it to Sentry. We should probably show an explicit error, but unsure yet what that should look like.
- Reporting to Sentry addresses the idea to do so in department-of-veterans-affairs/vets.gov-team#8930. We've probably also eliminated those errors about `then` not being a property of `undefined`, which seem related to the upfront URL fetch requests.

U-DON committed:
Capitalized SignIn.

U-DON committed:
Updated tests with minor changes to enable them.

U-DON committed:
Remove obsolete GA helpers.

U-DON committed:
Merge branch 'master' into login_routes

jbalboni commented on src/js/login/utils/helpers.js:
```diff
@@ -0,0 +1,65 @@
+      null,
+      ({ url }) => { popupWindow.location = url; },
+      () => { popupWindow.location = `${environment.BASE_URL}/auth/login/callback`; }
+    ).then(() => Promise.resolve(popupWindow));
```
You should be able to return popupWindow here without the Promise.resolve, I think.

jbalboni commented on src/js/login/utils/helpers.js:
```diff
@@ -0,0 +1,65 @@
+    extra: { url: popupUrl }
+  });
+
+  return Promise.reject('Failed to open new window');
```
In general, I think we should try to reject with Errors instead of strings.

jbalboni reviewed (approved):
This looks great. I tested it locally on the forms and didn't run into any issues. I had a couple minor comments, but this is so, so much better than what was there before.

dhall reviewed (approved):
This looks great! Thanks for all your work to simplify this as much as you have.

U-DON commented:
@jbalboni @saneshark @dhall Ready for review/test.

U-DON committed:
PR comments.

U-DON committed:
Empty commit.
